api/org/commcare/api/persistence/SqlIndexedStorageUtility.java

```diff
@@ -0,0 +1,388 @@
+/**
+ *
+ */
+package org.commcare.api.persistence;
+
+import org.javarosa.core.services.PrototypeManager;
+import org.javarosa.core.services.storage.EntityFilter;
+import org.javarosa.core.services.storage.IStorageUtilityIndexed;
+import org.javarosa.core.services.storage.Persistable;
+import org.javarosa.core.services.storage.StorageFullException;
+import org.javarosa.core.util.InvalidIndexException;
+import org.javarosa.core.util.externalizable.DeserializationException;
+import org.sqlite.javax.SQLiteConnectionPoolDataSource;
+
+import java.io.ByteArrayInputStream;
+import java.io.DataInputStream;
+import java.io.IOException;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+import java.util.Vector;
+
+/**
+ * IStorageIndexedUtility implemented on SQLite using JDBC. Contains all the functionality
+ * for interacting with the user's SQLite representation.
+ *
+ * @author wspride
+ */
+public class SqlIndexedStorageUtility<T extends Persistable> implements IStorageUtilityIndexed<T>, Iterable<T> {
+
+    private final Class<T> prototype;
+    private final String tableName;
+    private final String userName;
+
+    public SqlIndexedStorageUtility(Class<T> prototype, String userName, String tableName) {
+        this.tableName = tableName;
+        this.userName = userName;
+        this.prototype = prototype;
+        tryCreateTable();
+    }
+
+    Connection getConnection() throws SQLException, ClassNotFoundException {
+        Class.forName("org.sqlite.JDBC");
+        SQLiteConnectionPoolDataSource dataSource = new SQLiteConnectionPoolDataSource();
+        dataSource.setUrl("jdbc:sqlite:" + this.userName + ".db");
```

Comment:
If this is explicitly SQL-lite based I'd make that clear from the name.

Also: I don't think there's a specific reason to call this "username". Seems like "sandboxId" would be more clear.